controllers/keycloakrealmcomponent/keycloakrealmcomponent_controller.go

```diff
@@ -184,6 +190,10 @@ func (r *Reconcile) tryReconcile(
 		return fmt.Errorf("unable to create keycloak component: %w", err)
 	}
 
+	if err = r.secretRefClient.MapComponentConfigSecretsRefs(ctx, keycloakComponent.Config, keycloakRealmComponent.Namespace); err != nil {
```

Comment:
Let's modify `createKeycloakComponent`func to not modify the original object keycloakRealmComponent.

```go
func (r *Reconcile) createKeycloakComponent(
	ctx context.Context,
	component *keycloakApi.KeycloakRealmComponent,
	kcRealmName string,
	kClient keycloak.Client,
) (*adapter.Component, error) {
	ksComponent := &adapter.Component{
		Name:         component.Spec.Name,
		Config:       make(map[string][]string),
		ProviderID:   component.Spec.ProviderID,
		ProviderType: component.Spec.ProviderType,
	}

	for k, v := range component.Spec.Config {
		ksComponent.Config[k] = make([]string, len(v))
		copy(ksComponent.Config[k], v)
	}
	...
```

Reply:
Done